### PR TITLE
chore(docs): add prerequisite to install CStor-CSI on rancher based cluster

### PR DESCRIPTION
**Pull request by mittachaitu**
This PR adds prerequisite steps to install cStor-CSI
on rancher based Kubernetes cluster.

Signed-off-by: mittachaitu <sai.chaithanya@mayadata.io>

### CODE REVIEW

File: docs/troubleshooting/rancher_prerequisite.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ros config set rancher.services.user-volumes.volumes  [/home:/home,/opt:/opt,/var/lib/kubelet:/var/lib/kubelet,/etc/kubernetes:/etc/kubernetes,/var/openebs]
+system-docker rm all-volumes
+reboot
+```
+
+### iSCSI services on RHEL or Ubuntu
+
+#### Step1:  Verify iSCSI initiator is installed and services are running
+
+Below commands are required to verify and install iscsi services on nodes
```
Comment (prateekpandey14) [inline docs/troubleshooting/rancher_prerequisite.md:30]: As these iscsi installation and verification steps are common for all of them can we add one generic README with these steps as well .

Comment (mittachaitu) [reply docs/troubleshooting/rancher_prerequisite.md:30]: Doing it in a separate PR will be good.

Comment (mittachaitu) [reply docs/troubleshooting/rancher_prerequisite.md:30]: (Or) I will copy that tabular column after prerequisites.

File: docs/troubleshooting/troubleshooting.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ## FAQ and Troubleshooting
 
 1. My nodes in Kubernetes cluster were replaced and the cStor pools are pending. How can I fix this? See [Node Replacement](migrate_pool_by_migrating_disks.md).
-2. Pool manager pods are running but it is not reconciling the CSPC state. See [Bad Disk](pool_operations_hung.md)
\ No newline at end of file
+2. Pool manager pods are running but it is not reconciling the CSPC state. See [Bad Disk](pool_operations_hung.md)
+3. How to run cStor-CSI on rancher based clusters? See [Rancher Prerequisites](rancher_prerequisite.md)
```
Comment (sonasingh46) [inline docs/troubleshooting/troubleshooting.md:5]: Adding the rancher doc link here too will be good?

File: docs/troubleshooting/rancher_prerequisite.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+## Intro
+CStor-CSI provides block volume support through the iSCSI protocol. Therefore, the iSCSI client(initiator) presence on all Kubernetes nodes is required.
+
+## Step to be performed on Rancher based K8s cluster
+
+- If you are using RancherOS as the operating system, you need to enable the iSCSI service and start iSCSI service on all the worker nodes.
+- If you are using Ubuntu or RHEL as the operating system, you need to
+    - Verify if iSCSI initiators are installed on all nodes.
+    - Add the extra_binds under Kubelet service in cluster YAML file to mount the iSCSI binary and configuration inside the kubelet.
```
Comment (sonasingh46) [inline docs/troubleshooting/rancher_prerequisite.md:11]: Does this extra bind point need to go under RancherOS thing and not under Ubuntu?

File: docs/troubleshooting/troubleshooting.md (hunk 2 above)
Comment (mittachaitu) [reply docs/troubleshooting/troubleshooting.md:5]: IMO Yes, if user didn't follow the prerequisites properly and if things are not working as expected he will look at available troubleshooting section.

File: docs/troubleshooting/rancher_prerequisite.md (hunk 3 above)
Comment (mittachaitu) [reply docs/troubleshooting/rancher_prerequisite.md:11]: No, it should be under Ubuntu only. I followed [this](https://docs.openebs.io/docs/next/prerequisites.html#rancher) doc to fill these steps.

Comment (mittachaitu) [reply docs/troubleshooting/rancher_prerequisite.md:11]: I will verify by experimenting

Comment (sonasingh46) [reply docs/troubleshooting/rancher_prerequisite.md:11]: But I think I did not do any extra bind for ubuntu-based OS

Comment (mittachaitu) [reply docs/troubleshooting/rancher_prerequisite.md:11]: @sonasingh46 It is required for users who are using both SPC and CSPC APIs to provision volumes(If user doesn't perform extra binds then he will have problem for non-CSI based volumes).

File: docs/quick.md [hunk 4]
```diff
@@ -5,11 +5,26 @@
...
-1. You will need to have Kubernetes version 1.14 or higher
-2. iSCSI initiator utils installed on all the worker nodes
+1. You will need to have Kubernetes version 1.14 or higher.
+2. iSCSI initiator utils installed on all the worker nodes(If you are using rancher based cluster perform steps mentioned [here](troubleshooting/rancher_prerequisite.md)).
 3. You have access to install RBAC components into kube-system namespace.
 4. You have disks attached to nodes to provision storage.
 
+#### Install iSCSI initiator and load iSCSI modules
+
+| OPERATING SYSTEM | ISCSI PACKAGE         | COMMANDS                                                 |
```
Comment (kmova) [inline docs/quick.md:15]: Can you shift this table right below step 2 above?

File: docs/quick.md [hunk 5]
```diff
@@ -5,11 +5,26 @@
...
+| ---------------- | --------------------- | -------------------------------------------------------- |
+| RHEL/CentOS      | iscsi-initiator-utils | <ul><li>sudo yum install iscsi-initiator-utils -y</li><li>sudo systemctl enable --now iscsid</li><li>modprobe iscsi_tcp</li><li>echo iscsi_tcp >/etc/modules-load.d/iscsi-tcp.conf</li></ul> |
+| Ununtu/ Debian   | open-iscsi            |  <ul><li>sudo apt install open-iscsi</li><li>sudo systemctl enable --now iscsid</li><li>modprobe iscsi_tcp</li><li>echo iscsi_tcp >/etc/modules-load.d/iscsi-tcp.conf</li></ui>|
+| RancherOS        | open-iscsi            |  <ul><li>sudo ros s enable open-iscsi</li><li>sudo ros s up open-iscsi</li></ui>|
+
+- Verification of iscsi service:
```
Comment (kmova) [inline docs/quick.md:21]: Can this step be included next to the type of operating system in the above table.

File: docs/troubleshooting/rancher_prerequisite.md (hunk 1 above)
Comment (mittachaitu) [reply docs/troubleshooting/rancher_prerequisite.md:30]: updated

File: docs/quick.md (hunk 4 above)
Comment (mittachaitu) [reply docs/quick.md]: Updated

File: docs/quick.md (hunk 5 above)
Comment (mittachaitu) [reply docs/quick.md]: Updated

File: docs/troubleshooting/rancher_prerequisite.md (hunk 3 above)
Comment (sonasingh46) [reply docs/troubleshooting/rancher_prerequisite.md:11]: cc @gprasath @prateekpandey14

File: docs/troubleshooting/rancher_prerequisite.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+Run the below commands on all the nodes to make sure the below directories are persistent, by default these directories are ephemeral.
+```sh
+ros config set rancher.services.user-volumes.volumes  [/home:/home,/opt:/opt,/var/lib/kubelet:/var/lib/kubelet,/etc/kubernetes:/etc/kubernetes,/var/openebs]
+system-docker rm all-volumes
+reboot
+```
+
+### iSCSI services on RHEL or Ubuntu
+
+#### Step1:  Verify iSCSI initiator is installed and services are running
```
Comment (gprasath) [inline docs/troubleshooting/rancher_prerequisite.md:28]: Verify if iSCSI initiator is installed and services are running

File: docs/troubleshooting/rancher_prerequisite.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    - Verify if iSCSI initiators are installed on all nodes.
+    - Add the extra_binds under Kubelet service in cluster YAML file to mount the iSCSI binary and configuration inside the kubelet.
+
+### iSCSI services On RancherOS
+To run iSCSI services, execute the following commands on each of the cluster hosts or nodes.
+```sh
+sudo ros s enable open-iscsi
+sudo ros s up open-iscsi
+```
+Run the below commands on all the nodes to make sure the below directories are persistent, by default these directories are ephemeral.
```
Comment (gprasath) [inline docs/troubleshooting/rancher_prerequisite.md:19]: make sure that the below directories are

File: docs/troubleshooting/rancher_prerequisite.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
+# Prerequisites to run CStor-CSI in rancher based clusters
+
+## Intro
+CStor-CSI provides block volume support through the iSCSI protocol. Therefore, the iSCSI client(initiator) presence on all Kubernetes nodes is required.
+
+## Step to be performed on Rancher based K8s cluster
+
+- If you are using RancherOS as the operating system, you need to enable the iSCSI service and start iSCSI service on all the worker nodes.
+- If you are using Ubuntu or RHEL as the operating system, you need to
+    - Verify if iSCSI initiators are installed on all nodes.
```
Comment (gprasath) [inline docs/troubleshooting/rancher_prerequisite.md:10]: in ubuntu and rhel also, iscsi initiator would not be there by default. we need to install if it is not there

File: docs/troubleshooting/rancher_prerequisite.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+Below commands are required to verify and install iscsi services on nodes
+
+| OPERATING SYSTEM | ISCSI PACKAGE         | COMMANDS                                                 |
+| ---------------- | --------------------- | -------------------------------------------------------- |
+| RHEL/CentOS      | iscsi-initiator-utils | <ul><li>sudo yum install iscsi-initiator-utils -y</li><li>sudo systemctl enable --now iscsid</li><li>modprobe iscsi_tcp</li><li>echo iscsi_tcp >/etc/modules-load.d/iscsi-tcp.conf</li></ul> |
+| Ununtu/ Debian   | open-iscsi            |  <ul><li>sudo apt install open-iscsi</li><li>sudo systemctl enable --now iscsid</li><li>modprobe iscsi_tcp</li><li>echo iscsi_tcp >/etc/modules-load.d/iscsi-tcp.conf</li></ui>|
+
+#### Step2: Add extra_binds under kubelet service in cluster YAML
+
+After installing the initiator tool on your nodes, edit the YAML for your cluster, editing the kubelet configuration to mount the iSCSI binary and configuration, as shown in the sample below.
```
Comment (gprasath) [inline docs/troubleshooting/rancher_prerequisite.md:39]: After installing the iSCSI initiator on your nodes, bind them into the kubelet container by editing rancher `cluster.yaml`, as shown in the sample below.

File: docs/troubleshooting/rancher_prerequisite.md (hunk 3 above)
Comment (sonasingh46) [reply docs/troubleshooting/rancher_prerequisite.md:11]: I got it.

**Pull request merged**